app/docs/0.10.x/proxy.md

```diff
@@ -0,0 +1,759 @@
+[proxy-response]: #proxy-response
+[proxy-configuring-a-fallback-api]: #configuring-a-fallback-api
+[proxy-configuring-ssl-for-an-api]: #configuring-ssl-for-an-api
+[proxy-the-https-only-property]: #the-https_only-property
+[proxy-the-http-if-terminated-property]: #the-http_if_terminated-property
+[proxy-websocket]: #proxy-websocket-traffic
+[proxy-conclusion]: #conclusion
+
+---
+
+### Terminology
+
+In this document, we will refer to various Kong concepts. For the sake of 
+consistency, we will now define terms refering to such concepts:
+
+- `API`: This term refers to the API entity of Kong. You configure your APIs,
+that point to your own Upstream services through the Admin API.
+- `Plugin`: This refers to Kong "plugins", which are pieces of business logic
+that run in proxying lifecycle. Plugins can be configured globally (all 
+incoming traffic), or on a per-API basis through the Admin API.
+- `Client`: Refers to the - downstream - client making requests to Kong's proxy
+port.
+- `Upstream service`: Refers to your own API/service setting behind Kong,
+where requests are forwarded to.
+
+[Back to TOC](#table-of-contents)
+
+### Overview
+
+From a high level perspective, Kong will listen for HTTP traffic on its
+configured proxy port (`8000` by default), recognize which upstream service is
+being requested, run the configured plugins for that API, and forward the HTTP 
+request upstream to your own API or service. Kong can also run plugins once the 
+response from your upstream service is received, but that topic is not as
+relevant to this guide.
+
+When a client makes a request to the proxy port, Kong will decide where to route
+(or forward) that incoming request (that is, to which of your upstream 
+services). Kong achieves this through the APIs you configured (via the Admin 
+API). You can configure APIs with various properties, but the three relevant 
+ones for routing incoming traffic are `hosts`, `uris` and `methods`.
+
+If Kong cannot determine to which API a given request should be routed to, it
+will response with:
+
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+Server: kong/<x.x.x>
+
+{
+    "message": "no API found with those values"
+}
+```
+
+### Reminder: how to add an API to Kong
+
+Before goign any further, let's take a few moments to make sure you know how to
+add an API to Kong.
+
+As explained in the [Adding your API][adding-your-api] quickstart guide, Kong is
+configured via its internal [Admin API][API] running by default on port `8001`. 
+Adding an API to Kong is as easy as sending an HTTP request:
+
+```bash
+$ curl -i -X POST http://localhost:8001/apis/ \
+    -d 'name=my-api' \
+    -d 'upstream_url=http://my-api.com' \
+    -d 'hosts=my-api.com' \
+    -d 'uris=/my-api' \
+    -d 'methods=GET,HEAD'
+HTTP/1.1 201 Created
+...
+```
+
+This request instructs Kong to register an API named "my-api", reachable at
+"http://my-api.com". It also specifies various routing properties. In reality,
```

Comment:
It is idiomatic to use 'example.com' (or some thing RFC2606/6761 compliant hostname/tld) for cases such as this. Consider renaming cases of 'my-api.com' to 'example.com' or 'my-api.example'.